### DISCUSSION TITLE
Inconsistent behavior of breadcrumbs

### DISCUSSION THREAD

**Discussion by TryerGit** [microsoft/vscode-cpptools#8114] [Q&A]
I have a class implementation file with multiple functions defined. In VSCode, as I navigate the file, in the breadcrumbs view, some of the function names are displayed while others are not.

Please see this animated gif to see what I am referring to: https://i.stack.imgur.com/m9ABT.gif

For instance, anywhere inside function SV::checkconsistency(), the breadcrumbs are correctly displayed on top with the name of this outer function. This is true even when I am on line std::abort within this function.

When I then navigate down to function SV::Phase1(), breadcrumbs no longer displays the function name. Instead, it shows ... When I step on std::abort in this function, std::abort is displayed in breadcrumbs.

This is happening on VSCode 1.60.0 on ubuntu.

Should I be forcing generation of some symbols? This code compiles and runs fine. The above behavior occurs when the file is fully saved on disk.

**Comment by TryerGit** [comment]
I tested this out on Visual Studio IDE on Windows OS and there also only one of the functions appears in the breadcrumbs list and not the other. So, this does not seem to be a problem with VSCode alone or only with linux or Windows alone. Weird but rather persistent bug!

**Reply by sean-mcmanus** [reply]
It seems like some code is causing an issue with our tag parser. In particular, certain macros can break the tag parser and require entries to be added to a cpp.hint file, see https://docs.microsoft.com/en-us/cpp/build/reference/hint-files?view=msvc-160 . If you can provide repro code we could investigate more and/or you could file a bug on VS.
